src/HealthGPS/nutrients.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "HealthGPS.Core/identifier.h"
+
+#include "rapidcsv.h"
+
+#include <unordered_map>
+#include <vector>
+
+namespace hgps {
+namespace detail {
+/// @brief Work out which column each macronutrient isin
+/// @param column_names Column names for CSV file
+/// @param nutrients Macronutrients
+/// @return Index for each macronutrient in order
+std::vector<size_t> get_nutrient_indexes(const std::vector<std::string> &column_names,
+                                         const std::vector<core::Identifier> &nutrients);
+
+std::unordered_map<core::Identifier, std::vector<double>>
+get_nutrient_table(rapidcsv::Document &doc, const std::vector<core::Identifier> &nutrients,
+                   const std::vector<size_t> &nutrient_idx);
```

Comment:
Are these supposed to be internal functions used by `load_nutrient_table`? If so, would it be best to leave them out of the header?

Reply:
Yep. The only thing is that I have a unit test for `get_nutrient_indexes()`, so I need some way of accessing it. I suppose one way to do it would be to put the function declaration in the test file itself, though that's maybe a bit unorthodox. What do you think?

Reply:
Ah, good point. I used to have a separate header for internal functions, but that was C. I'd say it's fine as is then.